mlflow/tracking/fluent.py

```diff
@@ -1215,6 +1215,88 @@ def pagination_wrapper_func(number_to_get, next_page_token):
         )
 
 
+def search_runs_by_name(
```

Comment:
@r3stl355 Instead of introducing a new fluent API, can we add an `experiment_names` parameter to the existing `mlflow.search_runs()` API?

Reply:
Surely can, I was following @BenWilson2 suggestion to add a new API for this. Also, I think having a separate API makes it a bit cleaner, e.g. we'd then likely have to make some conditional checks and rules around what happens if both experiment_ids and experiment_names are provided. Let me know your preference and I can change it if needed

Reply:
We generally prefer to avoid expanding the fluent API surface unless necessary. Introducing separate methods also adds test complexity.

Reply:
Thanks for your flexibility here!

Reply:
And my profuse apologies on the misguided advice @r3stl355 . If it's any consolation, the implementation looked solid. 
As for the collision with both ids and names being provided, an exclusivity check on the existence of both in submitted args should raise an Exception to alert the user that what they think they're doing with submitting both won't actually do what they want it to do. It should be an either-or conditional for these args (with a notification in the signature docs and in the docs) instead of adding implementation complexity by figuring out a way to convert names to id's and then collapse the list to something that can be searched.

Reply:
Is this FR worth implementing at all?

Looking at the original reason this issue was raised, isn't adding `experiment_names` into `search_runs` with an additional constraint that only `experiment_names` or `experiment_ids` can be provided (but not both) in any way simpler than just leaving things as they are? (i.e. user needs to first call `mlflow.get_experiment_by_name` to get the `experiment_id` to use in a call to `search_runs`). It is going to be the same code behind the scene and there will be no performance improvements.

Reply:
That was the original thread of the comment on the initial feature request... "this really won't speed anything up". The request is more for simplifying a search_runs request to remove the additional step of having to get the experiment_ids directly from a get_experiment_by_name call. It's more of a convenience and code complexity reduction for an end user.

Reply:
Okey doke, new version on it's way 👍